apps/docs/content/guides/platform/migrating-and-upgrading-projects.mdx

```diff
@@ -93,6 +93,48 @@ If you upgrade to a paid plan while your project is paused, any expired one-clic
   src="/docs/img/guides/platform/paused-paid-tier.png"
 />
 
+#### Restoring a Downloaded Backup Locally
+
+Downloaded backups are plaintext SQL files that can be loaded into a Postgres instance using [psql](https://www.postgresql.org/docs/current/app-psql.html). You can use our open source build tooling if you would like to restore a backup into a local environment so you can interact with it. Here are the steps:
+
+Install the [nix](https://nixos.org/):
+
+```sh
+curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install
+```
+
+Next, we need to start postgres locally.
+
+If you know the project's Postgres version from your project's [infrastructure project settings](https://supabase.com/dashboard/project/_/settings/infrastructure) page, you can use it to spin up an identical image locally. Using an identical image improves the odds of successfully restoring your project because all of the extension versions and configuration will match exactly.
+
+Given Postgres version `15.6.1.121`, start postgres locally with the nix command:
+
+```sh
+nix run 'github:supabase/postgres?ref=15.6.1.127#start-server' 15
+```
+
+where `15` refers to the backup's Postgres major version.
+
+Note that the earliest Supabase Postgres version that supports a direct reference is `15.6.1.115`. If you don't know your project's Postgres version or it is lower than `15.6.1.115` then omit the `ref` option to build our most recent image:
+
+```sh
+nix run 'github:supabase/postgres#start-server' -- 15
+```
+
+Then we can restore the backup file into the Postgres instance from a separate shell session:
+
+```sh
+nix run github:supabase/postgres#start-client -- --file <BACKUP FILE>
+```
+
+Which loads the backup file into postgres and drops you into a [psql](https://www.postgresql.org/docs/current/app-psql.html) REPL. For `start-client` you do not need to provide a reference to the backup's Postgres version.
```

Comment:
I don't think most folks will know what an REPL is (I didn't!). maybe just a "psql prompt"